pkg/docker/docker.go

```diff
@@ -325,3 +341,29 @@ func logImagePullStatus(ctx context.Context, m *sync.Map) {
 
 	e.Msg("Pulling layers")
 }
+
+func getAuthToken(ctx context.Context, image string, dockerCreds config.DockerCredentials) string {
+	if dockerCreds.IsValid() {
+		// We only currently support auth for the default registry, so any
+		// pulls for `image` or `user/image` should be okay, anything trying
+		// to pull `repo/user/image` should not.
+		if strings.Count(image, "/") < 2 {
+			authConfig := types.AuthConfig{
+				Username: dockerCreds.Username,
+				Password: dockerCreds.Password,
+			}
+
+			encodedJSON, err := json.Marshal(authConfig)
+			if err != nil {
+				log.Ctx(ctx).Err(err).Msg("failed to encode docker credentials")
+			} else {
+				log.Ctx(ctx).Info().Msg("authenticated inspect from docker registry")
+				return base64.URLEncoding.EncodeToString(encodedJSON)
+			}
+		} else {
+			log.Ctx(ctx).Info().Msg("cannot authenticate for custom registry")
+		}
```

Comment:
Do custom registries require a different auth method??

Reply:
They use the same mechanism, but the structure that is jsonified and then base64-ified has different fields, including `ServerAddress`.  And in addition to username/password there are fields for other authentication credentials.  

We'd need a map of server -> creds for each value docker sees in the server part of `server/user/image` and that's probably not feasible.  

I think caching will be a much friendlier way of doing this for manifest info.